src/modules/ztsi/src/lib/Ztsi.cpp

```diff
@@ -313,40 +344,62 @@ int Ztsi::SetEnabled(bool enabled)
         if (enabled != configuration.enabled)
         {
             configuration.enabled = enabled;
-            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : MMI_OK;
+            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL;
         }
     }
     else if (ENOENT == status)
     {
         // If the configuration file doesn't exist, create it with the desired enabled state
         configuration.enabled = enabled;
-        configuration.serviceUrl = g_defaultServiceUrl;
-        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : MMI_OK;
+        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : EINVAL;
     }
 
     return status;
 }
 
-int Ztsi::SetServiceUrl(const std::string& serviceUrl)
+int Ztsi::SetMaxScheduledAttestationsPerDay(int maxScheduledAttestationsPerDay)
 {
     int status = MMI_OK;
-    AgentConfiguration configuration = {g_defaultServiceUrl, g_defaultEnabled};
+    AgentConfiguration configuration = {g_defaultEnabled, g_defaultMaxScheduledAttestationsPerDay, g_defaultMaxManualAttestationsPerDay};
+    status = ReadAgentConfiguration(configuration);
+    if ((MMI_OK == status) || (EINVAL == status))
+    {
+        if (maxScheduledAttestationsPerDay != configuration.maxScheduledAttestationsPerDay){
+            configuration.enabled = m_lastEnabledState;
+            configuration.maxScheduledAttestationsPerDay = maxScheduledAttestationsPerDay;
+            status = IsValidConfiguration(configuration) ? WriteAgentConfiguration(configuration) : EINVAL;
+        }
+    }
+    else if (ENOENT == status)
+    {
+        // If the configuration file doesn't exist, create it with the desired maxScheduledAttestationsPerDay state
+        configuration.enabled = m_lastEnabledState;
+        configuration.maxScheduledAttestationsPerDay = maxScheduledAttestationsPerDay;
+        status = IsValidConfiguration(configuration) ? CreateConfigurationFile(configuration) : EINVAL;
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

else?

Reply:
I'm not sure what the else condition would include in this case. We could log the error but it would already be getting logged in the call to ReadAgentConfiguration